trivia-app/Application/Categories/CategoriesViewControllerDelegate.swift

```diff
@@ -0,0 +1,33 @@
+//
+//  CategoriesViewControllerDelegate.swift
+//  trivia-app
+//
+//  Created by Rafael Villarreal on 15/01/20.
+//  Copyright © 2020 Rafael Villarreal. All rights reserved.
+//
+
+import UIKit
+
+extension CategoriesViewController: UICollectionViewDelegate, UICollectionViewDataSource {
+    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        return categories.count
+    }
+    
+    
+    func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
+        print(indexPath.item)
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        let cell: CategoryCell = collectionView.dequeueReusableCell(withReuseIdentifier: "CategoryCell", for: indexPath) as! CategoryCell
```

Comment:
Do not use as!, use as? and handle the case where it is nil, this is for preventing crashes.